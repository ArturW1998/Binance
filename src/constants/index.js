export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
