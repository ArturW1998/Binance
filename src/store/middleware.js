import logger from 'redux-logger';

const middleware = [];

if (process.env.NODE_ENV === 'development') {
  middleware.push(logger);
}

export default middleware;
