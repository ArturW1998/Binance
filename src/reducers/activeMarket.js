import { SET_ACTIVE_MARKET } from '../constants';

export const initState = {};

const activeMarket = (state = initState, { type, data }) => {
  switch (type) {
    case SET_ACTIVE_MARKET:
      return { ...data };
    default:
      return state;
  }
};

export default activeMarket;
