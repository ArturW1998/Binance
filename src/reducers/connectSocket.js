import { TOGGLE_SOCKET_STREAMS } from '../constants';

const initState = true;

const connectSocket = (state = initState, { type, status }) => {
  switch (type) {
    case TOGGLE_SOCKET_STREAMS:
      return status;
    default:
      return state;
  }
};

export default connectSocket;
