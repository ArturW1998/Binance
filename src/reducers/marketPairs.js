import { UPDATE_MARKET_PAIRS } from '../constants';

export const initState = {};

const marketPairs = (state = initState, { type, data }) => {
  switch (type) {
    case UPDATE_MARKET_PAIRS:
      return { ...state, ...data };
    default:
      return state;
  }
};

export default marketPairs;
