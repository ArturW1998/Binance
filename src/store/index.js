import { createStore, applyMiddleware, compose } from 'redux';
import { save, load } from 'redux-localstorage-simple';
import { CONFIG } from '../constants';
import rootReducer from '../reducers';
import middleware from './middleware';

const { LOCALSTORAGE_KEY } = CONFIG;

const composeEnhancers =
  process.env.NODE_ENV !== 'production' &&
  typeof window === 'object' &&
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose;

const savedState = save({ namespace: LOCALSTORAGE_KEY });

const configureStore = (preloadedState = {}) =>
  createStore(
    rootReducer,
    preloadedState,
    composeEnhancers(applyMiddleware(...middleware, savedState)),
  );

const preloadedState = load({ namespace: LOCALSTORAGE_KEY });

const store = configureStore(preloadedState);

export default store;
