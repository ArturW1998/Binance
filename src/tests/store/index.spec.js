import { ACTION_TYPES } from '../../constants';
import store from '../../store';
import activeMarket, {
  initState as activeMarketInitState,
} from '../../reducers/activeMarket';
import connectSocket, {
  initState as connectSocketInitState,
} from '../../reducers/connectSocket';
import marketPairs, {
  initState as marketPairsInitState,
} from '../../reducers/marketPairs';
import {
  toggleSocketStreams,
  updateMarketPairs,
  setActiveMarket,
} from '../../actions';

describe('store', () => {
  describe('store initial state', () => {
    it('initial state', () => {
      const expectedState = {
        activeMarket: activeMarket(activeMarketInitState, {}),
        connectSocket: connectSocket(connectSocketInitState, {}),
        marketPairs: marketPairs(marketPairsInitState, {}),
      };

      expect(store.getState()).toEqual(expectedState);
    });

    it('should have a store', () => {
      expect(store.getState()).toMatchSnapshot();
    });
  });

  describe('store toggleSocketStreams', () => {
    it('creates toggleSocketStreams when toggling socket streams has been done', () => {
      const { TOGGLE_SOCKET_STREAMS } = ACTION_TYPES;
      const expectedAction = {
        type: TOGGLE_SOCKET_STREAMS,
        status: false,
      };

      const { status } = expectedAction;

      store.dispatch(toggleSocketStreams(status));

      expect(store.getState().connectSocket).toEqual(
        connectSocket(connectSocketInitState, expectedAction),
      );
    });
  });

  describe('store updateMarketPairs', () => {
    it('creates updateMarketPairs when updating market pairs has been done', () => {
      const { UPDATE_MARKET_PAIRS } = ACTION_TYPES;
      const expectedAction = {
        type: UPDATE_MARKET_PAIRS,
        data: {
          BNBBTC: {},
          ETHUSDT: {},
          ENJBTC: {},
        },
      };

      const { data } = expectedAction;

      store.dispatch(updateMarketPairs(data));

      expect(store.getState().marketPairs).toEqual(
        marketPairs(marketPairsInitState, expectedAction),
      );
    });
  });

  describe('store setActiveMarket', () => {
    it('creates setActiveMarket when setting active market has been done', () => {
      const { SET_ACTIVE_MARKET } = ACTION_TYPES;
      const expectedAction = {
        type: SET_ACTIVE_MARKET,
        data: {
          market: 'BTC',
        },
      };

      const { data } = expectedAction;

      store.dispatch(setActiveMarket(data));

      expect(store.getState().activeMarket).toEqual(
        activeMarket(activeMarketInitState, expectedAction),
      );
    });
  });
});
