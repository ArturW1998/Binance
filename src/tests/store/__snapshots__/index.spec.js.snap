// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store store initial state should have a store 1`] = `
Object {
  "activeMarket": Object {},
  "connectSocket": true,
  "marketPairs": Object {},
}
`;
